packages/smart-wallet/src/smartWallet.js

```diff
@@ -928,10 +929,14 @@ export const prepareSmartWallet = (baggage, shared) => {
                 const purseP = facets.helper.purseForBrand(b);
 
                 // Now send it back to the purse.
-                return E(purseP).deposit(p);
+                return E(purseP)
+                  .deposit(p)
+                  .then(amt => out.push(amt));
               }),
             );
+            return harden(out);
```

Comment:
Should `brandPaymentRecord` stick around after this? All payments have been deposited, no?

Which makes me wonder what happens if some payments fail to deposit but not other.

I would think the following would be more robust:
- For each payment successfully deposited, remove from `brandPaymentRecord`
- If all payments have been successfully deposited, remove `brandPaymentRecord` from `liveOfferPayments`

```suggestion
            liveOfferPayments.delete(offerId);
            return harden(out);
```

This is of course assuming no other code path will mutate the `brandPaymentRecord` while depositing the payments, which I believe is the case as only `withdrawGive` seem to do that.

If handling partial successful deposits, maybe this method should return a `{amounts: Amount[], error?: Error | null}`, and never reject?

Reply:
re-visiting that logic is beyond the scope of this PR

IIUC, the `brandPaymentRecord` is deleted in `updateStatus`.

Reply:
I don't think updateStatus triggers in the path where reclaim payment does. But I may not fully understand the plumbing.

Reply:
right; updateStatus doesn't trigger until later. but it seems soon enough

Reply:
My claim is that it never triggers since there is no seat or offer to exit in this case.

Reply:
OK, I see now.

IOU an issue.